app/controllers/api/base_controller/authentication.rb

```diff
@@ -27,7 +33,10 @@ def require_api_user_or_token
           authenticate_with_system_token(request.headers[HttpHeaders::MIQ_TOKEN])
         when :token
           authenticate_with_user_token(request.headers[HttpHeaders::AUTH_TOKEN])
-        when :basic, :basic_ui, nil
+        when :cookie_ui
+          raise AuthenticationError unless valid_ui_session?
+          auth_user(session[:userid])
```

Comment:
I'm assuming the cookie check is in here?

Reply:
It actually happens earlier in the `valid_ui_session?` when validating the CSRF token as it is also part of the session.